lux-cli/src/build.rs

```diff
@@ -93,9 +93,11 @@ pub async fn build(data: Build, config: Config) -> Result<()> {
         if !build_dependencies_to_install.is_empty() {
             let bar = progress.map(|p| p.new_bar());
             luarocks.ensure_installed(&bar).await?;
-            Install::new(luarocks.tree(), luarocks.config())
+            // NOTE: Is this doing what we are expecting? Should it just
+            // install into the luarocks tree or should it install into the project?
```

Comment:
cc @mrcjkb i think i found a bug through this. The old code supplies the luarocks tree but then also supplies `.project()`, but all that ended up doing is using the project's tree :shrug:

Reply:
I think what we actually want here is a separate build-dependencies tree for the project (and maybe luarocks too?), like we have for test dependencies.

But we definitely shouldn't be installing build dependencies in the project tree.

Reply:
Resolved in #692.